tests/basic.test.tsx

```diff
@@ -147,6 +147,25 @@ it('only re-renders if selected state has changed', async () => {
   expect(controlRenderCount).toBe(1)
 })
 
+it('re-renders with useLayoutEffect', async () => {
+  const useEffect = jest.spyOn(React, 'useEffect')
+
+  const useStore = create(() => ({ state: false }))
+
+  function Test() {
+    const { state } = useStore()
+    React.useLayoutEffect(() => {
+      useStore.setState({ state: true })
+    }, [])
+    return <>{`${state}`}</>
+  }
+
+  const { container } = render(<Test />)
+
+  expect(container.innerHTML.trim()).toBe('true')
+  expect(useEffect).not.toHaveBeenCalled()
```

Comment:
Hm, this is testing the implementation details. In the future we might be using `useEffect` for different purpose.
Can we test the behavior? How about adding useEffect in `Test()` and make sure it's called just once?

Reply:
After searching a lot, I actually found how to properly test this behavior. The problem with react-testing-library is that it flushes all the useEffects and useLayoutEffects. Now it is testing the behavior.